### PR TITLE
Kill pod in kublet.syncPod if it is terminated

### PR DESCRIPTION
**Pull request by hex108**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Kill pod in kublet.syncPod if it is terminated(Failed or Succeed), then it avoids some unneeded operations(e.g. [`kl.makePodDataDirs`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1654) and [`kl.containerRuntime.SyncPod`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1674)).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/sig node

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1553,7 +1553,7 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	kl.statusManager.SetPodStatus(pod, apiPodStatus)
 
 	// Kill pod if it should not be running
-	if !runnable.Admit || pod.DeletionTimestamp != nil || apiPodStatus.Phase == v1.PodFailed {
+	if !runnable.Admit || pod.DeletionTimestamp != nil || kl.podIsTerminated(pod) {
```
Comment (mattjmcnaughton) [inline pkg/kubelet/kubelet.go]: It looks like there were some statements further down in this function that branched on `kl.podIsTerminated` - can we delete those if checks?

**Review by mattjmcnaughton — COMMENTED** [review]
Overall, I think this change makes sense, although I'd definitely be interested to get an opinion from another community member as there may be something I'm missing.

One question - is there unit test coverage for the `syncPod` function? Is it worth updating it to account for this new behavior?

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (hex108) [reply pkg/kubelet/kubelet.go]: Addressed.

**Comment by hex108** [comment]
> is there unit test coverage for the syncPod function? Is it worth updating it to account for this new behavior?

This PR just avoids some unneeded operations, it is not a new behavior, so it is a little hard to write a test to check it. :(

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Makes sense re testing.

**Comment by hex108** [comment]
/assign @Random-Liu @dchen1107 

for review

**Comment by hex108** [comment]
Kindly ping @Random-Liu @dchen1107

**Pull request closed without merging**